Source/NSGridView.m

```diff
@@ -0,0 +1,1280 @@
+{
+  NSArray *views = [self _viewsForColumnAtIndex: fromIndex];
+  [self removeColumnAtIndex: fromIndex];
+  [self insertColumnAtIndex: toIndex withViews: views];
+}
+
+- (void) removeColumnAtIndex: (NSInteger)index
+{
+  NSArray *objs = [self _cellsForColumnAtIndex: index];
+
+  [_columns removeObjectAtIndex: index];
+  [_cells removeObjectsInArray: objs];
+
+  [self _refreshCells];
+}
+
+- (NSGridCellPlacement) xPlacement
+{
+  return _xPlacement;
+}
+
+- (void) setXPlacement: (NSGridCellPlacement)x;
+{
+  _xPlacement = x;
+}
+
+- (NSGridCellPlacement) yPlacement;
+{
+  return _yPlacement;
+}
+
+- (void) setYPlacement: (NSGridCellPlacement)y;
+{
+  _yPlacement = y;
+}
+
+- (NSGridRowAlignment) rowAlignment;
+{
+  return _rowAlignment;
+}
+
+- (void) setRowAlignment: (NSGridRowAlignment)a;
+{
+  _rowAlignment = a;
+}
+
+- (CGFloat) rowSpacing
+{
+  return _rowSpacing;
+}
+
+- (void) setRowSpacing: (CGFloat)f
+{
+  _rowSpacing = f;
+}
+
+- (CGFloat) columnSpacing
+{
+  return _columnSpacing;
+}
+
+- (void) setColumnSpacing: (CGFloat)f
+{
+  _columnSpacing = f;
+}
+  
+- (void) mergeCellsInHorizontalRange: (NSRange)hRange verticalRange: (NSRange)vRange
+{
+  NSLog(@"Method %@ unimplemented.", NSStringFromSelector(_cmd));
+}
+
+// coding
+- (void) encodeWithCoder: (NSCoder *)coder
+{
+  [super encodeWithCoder: coder];
+  if ([coder allowsKeyedCoding])
+    {
+      [coder encodeInteger: _rowAlignment
+                    forKey: @"NSGrid_alignment"];
+      [coder encodeFloat: _columnSpacing
+                  forKey: @"NSGrid_columnSpacing"];
+      [coder encodeObject: _columns
+                   forKey: @"NSGrid_columns"];
+      [coder encodeFloat: _rowSpacing
+                  forKey: @"NSGrid_rowSpacing"];
+      [coder encodeObject: _rows
+                   forKey: @"NSGrid_rows"];
+      [coder encodeInteger: _xPlacement
+                    forKey: @"NSGrid_xPlacement"];
+      [coder encodeInteger: _yPlacement forKey: @"NSGrid_yPlacement"];
+      [coder encodeObject: _cells
+                   forKey: @"NSGrid_cells"];
+    }
+  else
+    {
+      [coder encodeValueOfObjCType:@encode(NSUInteger)
+                                at:&_rowAlignment];
+      [coder encodeObject: _columns];
+      [coder encodeObject: _rows];
+      [coder encodeObject: _cells];
+      [coder encodeValueOfObjCType:@encode(CGFloat)
+                                at:&_columnSpacing];
+      [coder encodeValueOfObjCType:@encode(CGFloat)
+                                at:&_rowSpacing];
+      [coder encodeValueOfObjCType:@encode(NSUInteger)
+                                at:&_xPlacement];
+      [coder encodeValueOfObjCType:@encode(NSUInteger)
+                                at:&_yPlacement];
+    }
+}
+
+- (instancetype) initWithCoder: (NSCoder *)coder
+{
+  self = [super initWithCoder: coder];
+  if (self != nil)
+    {
+      NSDebugLog(@"%@ %@",NSStringFromClass([self class]), NSStringFromSelector(_cmd));
```

Comment:
Before the merge most of these log statements need to be removed.

Reply:
Removed.  Only needed them while I was working on decoding.